core/amber/src/main/python/core/proxy/proxy_server.py

```diff
@@ -106,13 +106,13 @@ def __init__(
             description="Shut down this server.",
         )
 
-        # register control, this is the default action for the client to invoke
-        # after receiving control.
+        # register control, set default action for the client to invoke
+        # after receiving control.  it should invoke the control_handler defined
+        # in network_receiver and return number of batches in internal_queue to be
+        # used for credit calculation
         self.register(
             name="control",
-            action=ProxyServer.ack()(
-                lambda control_message: self.process_control(control_message)
-            ),
+            action=lambda control_message: self.process_control(control_message),
```

Comment:
can we just pass `self.process_control` as the action?

Reply:
@Yicong-Huang I tried doing this but got an error stating `process_control` is not an attribute of the class.  I believe at the time it binds this, self.process_control has not been assigned to the control_handler yet.